plugins/modules/tanzu_mission_control_secret.py

```diff
@@ -0,0 +1,778 @@
+  - This has only been tested with I(TMC SaaS), but it should also work with TMC Self Hosted.
+options:
+  api_host:
+    description:
+      - "Full URL of the TMC instance without the protocol, for example: V(my-organization.tmc.tanzu.broadcom.com)."
+    required: true
+    type: str
+  api_token:
+    description:
+      - API Token used to access the APIs.
+    required: true
+    type: str
+  management_cluster_name:
+    description:
+      - When creating a secret for a single cluster, this is the name of the management cluster.
+      - Required only when using O(cluster_name).
+      - Ignored when using O(cluster_group).
+    required: false
+    type: str
+  provisioner_name:
+    description:
+      - When creating a secret for a single cluster, this is the name of the provisioner.
+      - Required only when using O(cluster_name).
+      - Ignored when using O(cluster_group).
+    required: false
+    type: str
+  cluster_name:
+    description:
+      - Name of the existing cluster where the secret will be created.
+      - Mutually exclusive with O(cluster_group).
+    required: false
+    type: str
+  cluster_group:
+    description:
+      - Name of the existing cluster group where the secret will be created.
+      - Mutually exclusive with O(cluster_name).
+    required: false
+    type: str
+  cluster_namespace:
+    description:
+      - The namespace where the secret will be created.
+    required: true
+    type: str
+  secret_type:
+    description:
+      - Set this to V(opaque) to create an opaque secret (key/value pairs) or V(docker_config) to create a Docker Config secret.
+      - When O(secret_type=docker_config), O(registry_host), O(registry_username) and O(registry_password) are required.
+      - When O(secret_type=opaque), O(data) is required.
+    type: str
+    choices: ["opaque", "docker_config"]
+    default: opaque
+    required: false
+  registry_host:
+    description:
+      - Required when O(secret_type=docker_config), this is the Docker registry host.
+      - Mutually exclusive with O(data).
+    type: str
+    required: false
+  registry_username:
+    description:
+      - Required when O(secret_type=docker_config), this is the username of the Docker registry.
+      - Mutually exclusive with O(data).
+    type: str
+    required: false
+  registry_password:
+    description:
+      - Required when O(secret_type=docker_config), this is the password of the Docker registry.
+      - Mutually exclusive with O(data).
+    type: str
+    required: false
+  secret_name:
+    description:
+      - Name of the secret.
+    type: str
+    required: true
+  data:
+    description:
+      - Dictionary of key/value pairs for the secret.
+      - The values B(must not) be encoded in C(base64), the module will do that for you.
+      - Required when O(secret_type=opaque).
+      - Mutually exclusive with O(registry_host), O(registry_username) and O(registry_password).
```

Comment:
```suggestion
      - Mutually exclusive with O(registry_host), O(registry_username), and O(registry_password).
```
(https://docs.ansible.com/ansible/devel/dev_guide/style_guide/grammar_punctuation.html#commas)